src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/ContextQuery/SyntaxContext.cs

```diff
@@ -53,6 +56,15 @@ internal abstract class SyntaxContext
 
         public ImmutableArray<ITypeSymbol> InferredTypes { get; }
 
+        public bool IsInheritanceContext => IsBaseClassContext || IsBaseInterfaceContext || IsBaseRecordContext;
+
+        /// <summary>
+        /// When <see cref="IsInheritanceContext"/> is <see langword="true"/> contains syntax node of symbol inheriting from.
+        /// For instance for code snippet <c>class C : $$</c> it will contain ClassDeclarationSyntax for <c>C</c>.
+        /// Otherwise the value of this property is not defined (not guaranteed to be <see langword="null"/>)
+        /// </summary>
+        public SyntaxNode? DeclarationOfInheritingSymbol { get; }
```

Comment:
can use MemberNotNull attribute for this.

Reply:
tbh, i'd be fine with this *always* Being present.  and just call it `.ContainingTypeDeclaration`